scripts/assertions.bash

```diff
@@ -51,3 +51,13 @@ assert_tarball_not_contains_tags() { TARBALL="$1"; shift
 	tarball_tag_check_prep "$TARBALL" "$@"
 	assert_tags_do_not_exist_locally "$@"
 }
+
+assert_is_binfmt_registered() { 
+  test -f "/proc/sys/fs/binfmt_misc/$1"
+}
+
+assert_binfmt_fix_binary_flag_is_set() {
+  local flags
+  IFS=":" read _ flags <<< $(grep "flags" /proc/sys/fs/binfmt_misc/$1)
```

Comment:
What does the underscore do in `read _ ...` ? Would there be a case for adding the `-r` (raw) flag here? (Shellcheck has me trained up to always use that flag, but I'm sure there are cases when it's not appropriate.)

Reply:
I use that as `not used`, out of habit from other languages, though this question make me wonder if it had special meaning in bash. Which it does but it still probably safe since we don't use the value.  I'll update it to be a `local notused` though to make it more explicit and safe. 

I think it's the same either way with `-r` for this case. The line we're looking for is `flags: <FLAGS>` where `<FLAGS>` is some combination of `P`, `G`, `C`, and `F`, and there is only one `flags: ` line in that file.

Reply:
Ah, thanks! So we're reading two fields, not interested in the first, and expecting the second to be the flags field... I haven't seen read used in this way before, but makes sense now. Have you considered using the `cut` command to avoid needing to set IFS, something like this: ?

```bash
  flags="$(cut -d: -f2 <<< $(grep "flags" /proc/sys/fs/binfmt_misc/$1))"
```

I'm not sure if it's better or not, I like the explicitness of the `-f2` (field number 2) param and the `-d:` (delimiter colon) one, but now I understand read I think that's fine too.

Reply:
🤔  I hadn't thought of doing it like that.  `read` is more clear to me, but that's mostly due to familiarity, and I do like the explicitness of the cut,  changed 😄